blueoil/blueoil_train.py

```diff
@@ -20,17 +20,23 @@
 
 from blueoil.generate_lmnet_config import generate
 from executor.train import run as run_train
+from lmnet.utils import horovod as horovod_util
 
 
 def run(blueoil_config_file, experiment_id):
     """Train from blueoil config."""
 
-    # Copy bueoil config yaml.
-    output_dir = os.environ.get('OUTPUT_DIR', 'saved')
-    experiment_dir = os.path.join(output_dir, experiment_id)
-    save_config_file(blueoil_config_file, experiment_dir)
+    if horovod_util.is_enabled():
```

Comment:
I think this line is duplicated, you will call a `setup()` in below `is_rank0()`.

Reply:
Thanks for the comment, there must be a room to improvement.

There're some cases we need to consider:

1. horovod is not installed and user doesn't use horovod
2. horovod is installed but user doesn't use horovod
3. horovod is not installed but use try to use horovod (should be an error in this case)
4. horovod is installed and user try to use horovod

Current code is not well considered to handle these 4 cases, I guess. Also, another problem is that `is_rank0` should not call `setup`.

Reply:
I changed the code to exit immediately in the case of 3, since recover is impossible. I think now all cases are handled well.

Reply:
LGTM.